src/services/rolls/constants/index.ts

```diff
@@ -0,0 +1,211 @@
+const ALL_STATUS = ['Open', 'In Progress', 'Closed', 'Cancelled', 'Hold'] as const
+type StatusTuple = typeof ALL_STATUS
+type Status = StatusTuple[number]
+
+const ALL_SKILL = ['Art', 'Design', 'Engineering', 'Marketing', 'Business'] as const
+type SkillTuple = typeof ALL_SKILL
+type Skill = SkillTuple[number]
+
+const ALL_SKILL_LEVELS = ['Beginner', 'Intermediate', 'Advanced'] as const
+type SkillLevelsTuple = typeof ALL_SKILL_LEVELS
+type SkillLevel = SkillLevelsTuple[number]
+
+const ALL_ENTRY_REQUIREMENTS = ['Anyone', 'Restricted'] as const
+type EntryRequirementsTuple = typeof ALL_ENTRY_REQUIREMENTS
+type EntryRequirement = EntryRequirementsTuple[number]
+
+const ALL_BOUNTY_RANGES = [
+  [0, 500],
+  [500, 3000],
+  [3000, 100000],
+] as const
+type BountyRangesTuple = typeof ALL_BOUNTY_RANGES
+type BountyRange = BountyRangesTuple[number]
+
+const ALL_PROJECTS = ['Interface', 'MISO', 'Trident', 'Bento'] as const
+type ProjectsTuple = typeof ALL_PROJECTS
+type Project = ProjectsTuple[number]
+
+const REWARD_DENOMINATIONS = ['SUSHI', 'DAI', 'USDC'] as const
+type RewardDenominationsTuple = typeof REWARD_DENOMINATIONS
+type RewardDenomination = RewardDenominationsTuple[number]
+
+export interface BountyFilterParams {
+  status?: Status[]
+  skills?: Skill[]
+  skillLevels?: SkillLevel[]
+  entryRequirements?: EntryRequirement[]
+  bountyRange?: BountyRange
+}
+
+export interface Bounty {
+  id: number
+  project: Project
+  skill: Skill
+  skillLevel: SkillLevel
+  reward: number
+  rewardDenomination: RewardDenomination
+  createdAt: string
+  tags: string[]
+  entryRequirements: EntryRequirement
+
```

Comment:
This is something we may have to discuss. I think it will be easier to manage favorites if they are stored in relation to the user instead of the bounty object. So we can have a hook (ie useFavorites) in which we fetch the list of favorites for signed in user. Then we can cross reference that source for the prop. thoughts?

Reply:
yeah works with me